fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -129,26 +103,47 @@ public FedoraResource getResource(final Transaction transaction, final String id
     private FedoraResource instantiateResource(final Transaction transaction, final String identifier)
             throws PathNotFoundException {
         try {
-            final PersistentStorageSession psSession = getSession(transaction);
-            final ResourceHeaders headers = psSession.getHeaders(identifier, null);
+            final var psSession = getSession(transaction);
+            final var headers = psSession.getHeaders(identifier, null);
 
             // Determine the appropriate class from headers
-            final Class<? extends FedoraResource> createClass = getClassForTypes(headers);
+            final var createClass = getClassForTypes(headers);
 
             // Retrieve standard constructor
-            final Constructor<? extends FedoraResource> constructor = createClass.getConstructor(
-                    ResourceHeaders.class, Transaction.class, PersistentStorageSessionManager.class);
+            final var constructor = createClass.getConstructor(
+                    String.class,
+                    Transaction.class,
+                    PersistentStorageSessionManager.class,
+                    ResourceFactory.class);
+
+            final var rescImpl = constructor.newInstance(identifier, transaction,
+                    persistentStorageSessionManager, this);
+            populateResourceHeaders(rescImpl, headers);
 
-            return constructor.newInstance(headers, transaction, persistentStorageSessionManager);
+            return rescImpl;
         } catch (SecurityException | ReflectiveOperationException e) {
             throw new RepositoryRuntimeException("Unable to construct object", e);
         } catch (final PersistentItemNotFoundException e) {
             throw new PathNotFoundException(e);
-        } catch (PersistentStorageException e) {
+        } catch (final PersistentStorageException e) {
             throw new RepositoryRuntimeException(e);
         }
     }
 
+    private void populateResourceHeaders(final FedoraResourceImpl resc, final ResourceHeaders headers) {
```

Comment:
This method makes sense in the current mutable state of FedoraResourceImpl, but as (is my understanding) we are changing this to a more read-only interaction is this sustainable or will it have to come outside?

Reply:
Just to copy over what I said on slack:
this code is where fedora resources would be instantiated and populated, the properties have to get filled in at some point and the factory seems like the right place to me.
The resources should be read only after that